### PR TITLE
fix versioning issues

### PR DESCRIPTION
**Pull request by pekramp**
Issue when trying to use OpenShift 3.10/3.11 or above original logic assumed only two digits i.e. 3.7, 3.9

### CODE REVIEW

File: disconnected_registry/docker-registry-sync.py [hunk 1]
```diff
@@ -84,16 +84,20 @@ def get_latest_tag_from_api(url_list, tag_list, failed_image_list, version_type
             sys.exit()
         # Get the latest version for a given release
         latest_tag = ''
+	temp_tag = ''
```
Comment (Cameronwyatt) [inline disconnected_registry/docker-registry-sync.py]: Fix spacing

File: disconnected_registry/docker-registry-sync.py [hunk 2]
```diff
@@ -84,16 +84,20 @@ def get_latest_tag_from_api(url_list, tag_list, failed_image_list, version_type
...
+        req_tag = ''
         image_name = image_tag_dictionary['name']
         for tag in image_tag_dictionary['tags']:
             # check to see if there is a 'v' in the version tag:
             if tag.startswith('v'):
-                # This tracks the position of the splice. It assumes that you are trying to get the latest
-                # release based on a two digit release (i.e. 3.4 or 3.7)
-                splice_position = 4
+               # Ensures taht a valid version is being parsed and searched for
```
Comment (Cameronwyatt) [inline disconnected_registry/docker-registry-sync.py]: Type `taht`

File: disconnected_registry/docker-registry-sync.py [hunk 3]
```diff
@@ -84,16 +84,20 @@ def get_latest_tag_from_api(url_list, tag_list, failed_image_list, version_type
...
+               # release based on a two series release (i.e. 3.9 or 3.10)
+              temp_tag = tag.split('.')
+              req_tag = temp_tag[0] +'.'+ temp_tag[1]
```
Comment (Cameronwyatt) [inline disconnected_registry/docker-registry-sync.py]: `req_tag = ".".join(temp_tag[:2]`

File: disconnected_registry/docker-registry-sync.py [hunk 4]
```diff
@@ -108,12 +112,19 @@ def get_latest_tag_from_api(url_list, tag_list, failed_image_list, version_type
...
             logging.error("Are you sure that the version exists in the RedHat registry?")
-            failed_image_list.append(image_name)
+            logging.info("Attempting to pull image tag 'latest' instead")
+            #insted of failing we will try to pull the tag 'latest'
+            #failed_image_list.append(image_name)
+            tag_list.append("%s:%s" % (image_name, 'latest'))
         else:
             tag_list.append("%s:%s" % (image_name, latest_tag_minus_hyphon))
 
+        #If package is an rhgs3 package grabe the version and 'latest', discovered trying to deploy 3.10 disconnected
```
Comment (Cameronwyatt) [inline disconnected_registry/docker-registry-sync.py]: type "grabe"

**Comment by Cameronwyatt** [comment]
Please also squash the commits

**Comment by sabre1041** [comment]
@Cameronwyatt have you validated the changes in this PR? We can have GH squash the commits if necessary

**Comment by etsauer** [comment]
@Cameronwyatt @pekramp @sabre1041 this has been open for a little while. Can we get it merged?

File: disconnected_registry/docker-registry-sync.py (hunk 3 above)
Comment (pekramp) [reply disconnected_registry/docker-registry-sync.py]: fixed

File: disconnected_registry/docker-registry-sync.py (hunk 4 above)
Comment (pekramp) [reply disconnected_registry/docker-registry-sync.py]: fixed

**Comment by pekramp** [comment]
> @Cameronwyatt @pekramp @sabre1041 this has been open for a little while. Can we get it merged?

completed Cameron's recommended fixes. This has been tested on our disconnected install for a client with Cameron and Nick Poyant

**Pull request merged**